### PR TITLE
feat(net): stabilize Deno.listenDatagram with udp

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

File: cli/tsc/dts/lib.deno.unstable.d.ts [hunk 1]
```diff
@@ -892,70 +893,6 @@ declare namespace Deno {
...
-    joinMulticastV4(
-      address: string,
-      networkInterface: string,
-    ): Promise<MulticastV4Membership>;
-
-    /** Joins an IPv6 multicast group. */
-    joinMulticastV6(
-      address: string,
-      networkInterface: number,
-    ): Promise<MulticastV6Membership>;
```
Comment (mmastrac) [inline cli/tsc/dts/lib.deno.unstable.d.ts:940]: I wonder if this should be split out to a `UDPDatagramConn` as this `DatagramConn` interface would be useful to support datagrams-over-QUIC (which won't support multicast).

Comment (crowlKats) [reply cli/tsc/dts/lib.deno.unstable.d.ts:940]: I think it could just throw on those methods

**Comment by kt3k** [comment]
Sorry for bringing this up very late, but is `listen` correct verb for this API? `listen` sounds like it starts a server, but `DatagramConn` works as both client and server depending on the later API calls (`send` or `receive`)

Node.js calls the similar API `dgram.createSocket` which seems expressing its dual (`sender` & `receiver`) nature more naturally. What do you think?

**Comment by lino-levan** [comment]
Definitely agree with @kt3k on this one. The naming felt really strange to me as I was writing that example. Something like `createDatagram` or `createSocket` would be a lot clearer.

File: ext/net/lib.deno_net.d.ts [hunk 2]
```diff
@@ -309,6 +309,106 @@ declare namespace Deno {
...
+    receive(p?: Uint8Array): Promise<[Uint8Array, Addr]>;
+    /** Sends a message to the target via the connection. The method resolves
+     * with the number of bytes sent. */
+    send(p: Uint8Array, addr: Addr): Promise<number>;
+    /** Close closes the socket. Any pending message promises will be rejected
+     * with errors. */
+    close(): void;
+    /** Return the address of the instance. */
+    readonly addr: Addr;
+    [Symbol.asyncIterator](): AsyncIterableIterator<[Uint8Array, Addr]>;
```
Comment (aapoalas) [inline ext/net/lib.deno_net.d.ts:409]: **question:** Given that we already expose an async iterator of the connection, what is actually the thing that stops us from making this into an RW stream API? Is it the buffer + address tuple?

Comment (crowlKats) [reply ext/net/lib.deno_net.d.ts:409]: it would be a possibility to make that a readable stream

Comment (lino-levan) [reply ext/net/lib.deno_net.d.ts:409]: How could this be a stream given the address information? This doesn't seem super clear to me.

**Comment by bartlomieju** [comment]
I'm gonna close this PR for now, since there's no consensus in the core team about stabilizing this API in its current form.

**Pull request closed without merging**